source/opt/passes.h

```diff
@@ -74,6 +80,142 @@ class EliminateDeadConstantPass : public Pass {
+  ir::Instruction* DoComponentWiseOperation(
+      ir::Module::inst_iterator* inst_iter_ptr);
+
+  // Creates a constant defining instruction for the given Const instance and
+  // inserts the instruction at the position specified by the given instruction
+  // iterator. Returns a pointer to the created instruction if succeeded,
+  // otherwise returns a null pointer. The instruction iterator points to the
+  // same instruction before and after the insertion. This is the method that
+  // actually manages new id creation and assignment.
+  ir::Instruction* BuildInstructionAndAddToModule(
+      std::unique_ptr<analysis::Const> c, ir::Module::inst_iterator* pos);
+
+  // Creates a Const instance to hold the constant value of the given
+  // instruction. If the given instruction defines a normal constants whose
+  // value is already known in the module, returns the unique pointer to the
+  // created Const instance. Otherwise does not create anything and returns a
+  // nullptr.
+  std::unique_ptr<analysis::Const> CreateConstFromInst(ir::Instruction* inst);
+
+  // Creates a Const instance with the given type and a vector of constant
+  // defining words. Returns an unique pointer to the create Const instance if
+  // the Const instance can be created successfully. Otherwise returns a null
+  // pointer. To create scalar type constants, the vector should contain the
+  // constant value in 32 bit words. To create composite type constants, the
+  // vector should contain the component ids. To create null constants, the
+  // vector should be empty.
+  std::unique_ptr<analysis::Const> CreateConst(
+      const analysis::Type* type,
+      const std::vector<uint32_t>& literal_words_or_ids);
+
+  // Creates an instruction with the given result id to declare a constant
+  // represented by the given Const instance. Returns an unique pointer to the
+  // created instruction if the instruction can be created successfully.
+  // Otherwise, returns a null pointer.
+  std::unique_ptr<ir::Instruction> CreateInstruction(uint32_t result_id,
+                                                     analysis::Const* c);
+
+  // Creates an OpConstantComposite instruction with the given result id and
+  // the CompositeConst instance which represents a composite constant. Returns
+  // an unique pointer to the created instruction if succeeded. Otherwise
+  // returns a null pointer.
+  std::unique_ptr<ir::Instruction> CreateCompositeInstruction(
+      uint32_t result_id, analysis::CompositeConst* cc);
+
+  // A helper function to get the collected normal constant with the given id.
+  // Returns the pointer to the Const instance in case it is found. Otherwise,
+  // returns null pointer.
+  analysis::Const* FindRecordedConst(uint32_t id);
+  // A helper function to get the id of a collected constant with the pointer
+  // to the Const instance. Returns 0 in case the constant is not found.
+  uint32_t FindRecordedConst(const analysis::Const* c);
```

Comment:
Yes. This function simply looks up `c` in the `const_val_to_id_` map. When `c` is nullptr, it should return 0, which means an invalid id.